plugins/techdocs-backend/src/service/DocsBuildStrategy.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+
+export interface DocsBuildStrategy {
+  shouldBuild(entity: Entity): Promise<boolean>;
+}
+
+export class DefaultDocsBuildStrategy {
+  private readonly config: Config;
+
+  constructor(config: Config) {
```

Comment:
Would you say this is a candidate for `fromConfig` static method with a private/no constructor too? @backstage/maintainers